### ISSUE TITLE
require only scans the first directory in package.path

### ISSUE THREAD

**Issue by jeychenne** [ThePhD/sol2#513]
I'm trying to load modules from standard paths in `package.path` on linux. Everything works fine with the standalone version of lua, but I'm having a problem with  sol/embedded lua. Here is the content of my  `package.path` :  

```
/usr/local/share/lua/5.3/?.lua;/usr/local/share/lua/5.3/?/init.lua;/usr/local/lib/lua/5.3/?.lua;/usr/local/lib/lua/5.3/?/init.lua;./?.lua;./?/init.lua
```

Now, if I put a `hello.lua` module in `/usr/local/lib/lua/5.3/`, I get the following error:

`sol: ok error:  /usr/local/share/lua/5.3/hello.lua`

If I put the script in `/usr/local/share/lua/5.3/`, then it works. The same thing happens if I customize `package.path` with my own directories (it works if it's in the first directory listed, otherwise it doesn't). 

Does anyone have any idea why I get this weird behavior? Am I missing something obvious? Thanks in advance...

**Comment by ThePhD** [comment]
I, personally, have no idea how to help you. I don't really use `package.path`. If it works in the standalone lua executable, perhaps look at Lua 5.3's "lua.c" source file to see how it works with package.path / configures package stuff.

**Comment by squeek502** [comment]
@jeychenne could you share a bit more about the specifics of what you're trying? What script is being loaded by sol, and how is it being loaded? How/where is package.path being set? Is that the full error? Just a filename as an error seems strange.

**Comment by jeychenne** [comment]
Thanks to both of you. I finally found out what the problem was, and it was neither Lua's nor sol's fault... 
As it turns out, I use a slightly modified version of Lua where I have replaced calls to `fopen` and `freopen` with a version that handle Unicode paths on Windows. The problem was that I was throwing an exception when I got a NULL file handle, instead of returning whatever I got and letting Lua do it's thing. Now it's fixed...
